--- a/src/libcec/cmake/CheckPlatformSupport.cmake	2017-07-18 18:43:30.000000000 +0200
+++ b/src/libcec/cmake/CheckPlatformSupport.cmake	2017-08-22 14:04:36.922726999 +0200
@@ -190,10 +190,10 @@
     swig_link_libraries(cec cec)
 
     SET(PYTHON_LIB_INSTALL_PATH "/cec" CACHE STRING "python lib path")
-    if (${CMAKE_MAJOR_VERSION} GREATER 2 AND ${CMAKE_MAJOR_VERSION} GREATER_EQUAL 7)
-	  SET(PYTHON_LIB_INSTALL_PATH "" CACHE STRING "python lib path" FORCE)
-    else()
+    if (${CMAKE_MAJOR_VERSION} GREATER 2 AND ${CMAKE_MAJOR_VERSION} GREATER 6)
+	  SET(PYTHON_LIB_INSTALL_PATH "" CACHE STRING "python lib path" FORCE)
+    else()
-      if (${CMAKE_MAJOR_VERSION} GREATER_EQUAL 3)
+      if (${CMAKE_MAJOR_VERSION} GREATER 2)
         SET(PYTHON_LIB_INSTALL_PATH "" CACHE STRING "python lib path" FORCE)
       endif()
     endif()
--
