--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2017-04-21 10:59:59.209075420 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp	2017-04-21 11:06:02.641075420 +0200
@@ -638,7 +638,7 @@
           break;
       }
 
-      if (framerate > 0.0 && (int)m_framerate != (int)framerate)
+      if (framerate > 100.0 && fabs(m_framerate - framerate) > 1.0)
       {
         m_framerate = framerate;
         m_video_rate = (int)(0.5 + (96000.0 / framerate));
@@ -648,8 +648,8 @@
 
         m_processInfo.SetVideoFps(m_framerate);
 
-        CLog::Log(LOGDEBUG, "%s: detected new framerate(%f), video_rate(%d)",
-          __MODULE_NAME__, m_framerate, m_video_rate);
+        CLog::Log(LOGDEBUG, "%s: detected new framerate(%f), video_rate(%d), dur:%d",
+          __MODULE_NAME__, m_framerate, m_video_rate, (int)(0.5 + duration));
       }
     }
 
--
